### PR TITLE
RWA-1669 execute reconfiguration will make use of canReconfigure attribute from Configuration DMN

### PR DESCRIPTION
**Pull request by vgulapally**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-1669

### Change description ###
execute reconfiguration will use Configuration DMN's can reconfigure attribute before reconfiguring task management task attributes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java [hunk 1]
```diff
@@ -474,27 +493,101 @@ private void mapVariableToTaskResourceProperty(TaskResource taskResource, String
...
+
+    void setMinorPriority(Object value, TaskResource taskResource) {
+        if (value instanceof String) {
+            taskResource.setMinorPriority(Integer.parseInt((String) value));
+        } else {
+            taskResource.setMinorPriority((Integer) value);
+        }
+    }
+
+    protected void reconfigureTaskAttribute(TaskResource taskResource,
```
Comment (Lee365) [inline src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java:530]: reconfigureTaskAttribute method seems to have code duplicated from mapVariableToTaskResourceProperty. Is it possible to combine these two methods with check on reconfigure?

Comment (vgulapally) [reply src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/CFTTaskMapper.java:530]: they both have different set of properties, don't want to mess with existing logic

**Review by Lee365 — APPROVED** [review]
@JsonBackReference on TaskRoleResource can be removed if no longer required after reconfigure code change.

**Comment by vgulapally** [comment]
> @JsonBackReference on TaskRoleResource can be removed if no longer required after reconfigure code change.

It needed mate

**Pull request merged**